jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/ToolkitAuthManager.kt

```diff
@@ -143,6 +144,18 @@ fun loginSso(project: Project?, startUrl: String, scopes: List<String> = ALL_SON
     }
 }
 
+fun logoutFromSsoConnection(project: Project?, connection: ToolkitConnection, callback: () -> Unit = {}) {
+    if (connection !is AwsBearerTokenConnection) return
+    ApplicationManager.getApplication().messageBus.syncPublisher(BearerTokenProviderListener.TOPIC).invalidate(connection.id)
+    ToolkitAuthManager.getInstance().deleteConnection(connection.id)
+
+    project?.let {
+        ToolkitConnectionManager.getInstance(it).switchConnection(null)
+    }
+
+    callback()
```

Comment:
Callback never gets executed if the previous statements throw. This is somewhat surprising IMO.

Reply:
ack make sense

Reply:
but have 1 qq: since here we don't have any API call, can't we expect or assume it will always succeed?

Reply:
> can't we expect or assume it will always succeed?

No, we should never assume that a statement will succeed. Every line of imperative logic is directly dependent on the success of the preceding line. If you want to guarantee execution then a try/finally is needed.

Reply:
gotcha thanks!